server/src/controllers/middleware/getFavorite.js

```diff
@@ -0,0 +1,34 @@
+const { getFavoriteCourse, getUserById } = require('../../database/queries');
+const { favoriteSchema } = require('../../utils');
+
+const getFavorite = async (req, res, next) => {
+  try {
+    const { userId } = req.params;
+    await favoriteSchema.validate({ userId });
+    if (userId > 0) {
+      const checkUser = await getUserById(userId);
+      if (checkUser.rowCount) {
+        const { rows } = await getFavoriteCourse(userId);
+        if (rows[0]) {
+          res.json(rows);
+        } else {
+          res.json({
+            message: 'There is no favorite courses for this user id',
+          });
+        }
+      } else {
+        res.status(404).json({
+          message: 'Invalid id',
+        });
```

Comment:
**Really, you did a good job in handling all cases here :+1:**  but, there is a small point : 
* `if (rows[0]) ` , here you want to check if the rows is empty or not so, you can use `rows.length `instead

Reply:
Thank you,
But here will not be any difference in the two case

Reply:
If row is empty row[0] will be undefined so the condition will not pass

Reply:
yes, no difference but at the first shot I think you checked for rows[0] not for the length